docs/bootstrap.sh

```diff
@@ -67,40 +70,90 @@ function release_preview {
   fi
 }
 
-function release {
-  echo_header "docs release"
+function docs_cut_version {
+    echo_header "docs version"
 
-  # If we download cached docs, we may not have netlify CLI in node_modules. Install in case.
-  yarn install
+    if [ -z "$1" ]; then
+        echo "Usage: $1 <version>"
+        exit 1
+    fi
+
+    # Store the current branch to return to later
+    current_branch=$(git rev-parse --abbrev-ref HEAD)
+    echo "Original branch: $current_branch"
+
+    COMMIT_TAG=$1
+    echo "Starting docs versioning for $COMMIT_TAG"
+
+    # Stash any unrelated local changes before starting
+    echo "Stashing local changes"
+    stash_message="docs-version-stash-$(date +%s)"
+    git stash push -m "$stash_message"
+    local stashed=$? # Check if something was stashed
+
+    rm -rf processed-docs processed-docs-cache
+
+    echo "Processing tag: $COMMIT_TAG"
+
+    # Checkout the tag, discarding local changes from previous build artifacts
+    # Use --force to overwrite potentially modified tracked files from the build process
+    echo "Checking out tag $COMMIT_TAG..."
+    if ! git checkout --force "$COMMIT_TAG"; then
+        echo "Error checking out tag $tag. Aborting."
+        # Go back to original branch and restore stash before exiting
+        git checkout "$current_branch"
+        if [ $stashed -eq 0 ]; then git stash pop; fi
+        exit 1
+    fi
+
+    # Prepare for docusaurus build/versioning for this tag
+    echo "[]" > versions.json # Docusaurus versioning might need this cleared
+    echo "Running preprocess and build for $COMMIT_TAG..."
+    yarn preprocess # Assuming this doesn't modify tracked files in a conflicting way
+    yarn docusaurus build # This might modify tracked files
+
+    # Create the versioned docs for this tag
+    echo "Creating documentation version for $COMMIT_TAG..."
+    # Pass COMMIT_TAG env var specifically if needed by the command
+    if ! COMMIT_TAG=$COMMIT_TAG yarn docusaurus docs:version "$COMMIT_TAG"; then
```

Comment:
here's the actual version cutting, need to set `COMMIT_TAG` because of the `#include_aztec_version` macro